### PR TITLE
feat: support custom docker images

### PR DESCRIPTION
**Pull request by andrewthauer**
## Hey, I just made a Pull Request!

This adds new `--docker-image` command line switches to support a custom docker image. This can be useful if you want to support custom MkDocs plugins or external tools.

#### :heavy_check_mark: Checklist

- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: packages/techdocs-cli/src/commands/index.ts [hunk 1]
```diff
@@ -100,6 +108,10 @@ export function registerCommands(program: CommanderStatic) {
     .description(
       "Serve a documentation project locally in a Backstage app-like environment"
     )
+    .option(
```
Comment (iamEAP) [inline packages/techdocs-cli/src/commands/index.ts:111]: Recommend tacking on the `spotify/techdocs` default value to these options declarations for backwards compatibility (believe the third argument is automatically provided as the default value to the underlying command without having to specify it there).

**Review by iamEAP — COMMENTED** [review]
Nice. Thanks @andrewthauer!  This approach makes sense to me.  One little comment about defaults.

Something to think about: I wonder if there are security implications for a change like this?

File: packages/techdocs-cli/src/commands/index.ts (hunk 1 above)
Comment (andrewthauer) [reply packages/techdocs-cli/src/commands/index.ts:108]: Updated

**Comment by andrewthauer** [comment]
@iamEAP - I've decoupled this from the https://github.com/backstage/backstage/pull/6071 since the `serve` functionality can be used independently from generate (which can be accomplished already with `local` or custom code). I'll do a follow up to support `generate --docker-image` once https://github.com/backstage/backstage/pull/6071 is finalized.

**Review by iamEAP — APPROVED** [review]
Nice!  Thanks @andrewthauer.  My only last thought is that we should probably document this somewhere (e.g. it's not 100% clear that the custom docker image must respond to the same args as the default container).  ...But maybe the `backstage.io` docs are a better spot for that.

I'll leave it to @camilaibs to merge.

**Comment by camilaibs** [comment]
> Nice! Thanks @andrewthauer. My only last thought is that we should probably document this somewhere (e.g. it's not 100% clear that the custom docker image must respond to the same args as the default container). ...But maybe the `backstage.io` docs are a better spot for that.
> 
> I'll leave it to @camilaibs to merge.

Hey @andrewthauer and @iamEAP, I'll merge tomorrow!

**Comment by andrewthauer** [comment]
> > Nice! Thanks @andrewthauer. My only last thought is that we should probably document this somewhere (e.g. it's not 100% clear that the custom docker image must respond to the same args as the default container). ...But maybe the `backstage.io` docs are a better spot for that.

This is a very good point. Yes, the custom container would definitely need to have `mkdocs` as the entry point to work I would think. I'll take a look at documenting this.

**Comment by andrewthauer** [comment]
I've updated the `docs/README.md` file. Can also look into doing this for `backstage.io` as well.

**Pull request merged**